### PR TITLE
Apply changes to ADRIA to make it compatible with CoralBlox model

### PR DESCRIPTION
**Pull request by Zapiano**
There is a very long list of changes that were made to ADRIA to replace the ODE approach by CoralBlox to take care of coral growth and base mortality. Following there's a list of relevant things, please feel free to update it:

- Now ADRIA is working with 5 functional groups and 7 size classes
  - Update `linear_extensions`, `bin_widths` and other params accordingly
  - Update some metrics and analysis that were expecting a 6x6 or a vector with species dimension of length 36
- Update `run_scenario`, `run_scenario` and `run_model` to use CoralBlox at each timestep
- Expect the ADRIA domain datapackage initial coral cover to **not be** split into size classes, now that is happening when `load_domain`
  - Bump expected version of the datapackage to `v0.7.0-rc` 
- Update tests

There are some defunct functions that I intend to remove on a separate PR. 
Documentation should also be update, again I intend to do that on a separate PR.

### CODE REVIEW

**Comment by Zapiano** [comment]
@DanTanAtAims can you rebase this again, please? Then I can do a last review.

**Review by Zapiano — COMMENTED** [review]
@DanTanAtAims I left two comments that I think you can help answer.

File: src/ecosystem/corals/Corals.jl [hunk 1]
```diff
@@ -53,19 +121,25 @@ Generate colony area data based on Bozec et al., [1].
...
     # To convert to cover we locate bin means and calculate bin mean areas
-    colony_diam_means = repeat(mean_cm_diameters', nclasses, 1)
-    colony_area_mean_cm2 = colony_mean_area(colony_diam_means)
+    colony_area_mean_cm2 = colony_mean_area(mean_cm_diameters)
 
-    return colony_area_mean_cm2, (colony_diam_means ./ 100.0)
+    return colony_area_mean_cm2, linear_scale.(mean_cm_diameters, :cm, :m)
+end
+
+function bins_bounds(mean_diam::Matrix{Float64})::Matrix{Float64}
```
[the remark this answers is not in the document]
Comment (ConnectedSystems) [reply src/ecosystem/corals/Corals.jl:136]: This looks to be an interface function to allow us to extract the bin bounds easily.

If we're not using it internally, is it still nice to have?

If not, agree that it should be removed.

File: src/ecosystem/corals/Corals.jl [hunk 2]
```diff
@@ -127,115 +201,109 @@ function coral_spec()::NamedTuple
...
+    # Mortality
+    # Survival rates taken from (unpublished) ecoRRAP data.
+    # Background mortality is then the complement.
+    # survival_rate::Matrix{Float64} = [
+    #     0.859017851 0.858528906 0.857044217 0.856477498 0.856104353 0.855852241 0.855852241;    # Tabular Acropora
+    #     0.865006527 0.87915437 0.892044073 0.905304164 0.915373252 0.925707536 0.925707536;     # Corymbose Acropora
+    #     0.953069031 0.959152694 0.964460394 0.968306361 0.972598906 0.97621179 0.97621179;     # Corymbose non-Acropora
+    #     0.869976692 0.938029324 0.977889252 0.987199004 0.99207702 0.996931548 0.996931548;     # Small massives and encrusting
+    #     0.9782479 0.979496637 0.980850254 0.982178103 0.983568572 0.984667677 0.984667677       # Large massives
+    # ]
```
Comment (ConnectedSystems) [inline src/ecosystem/corals/Corals.jl:266]: Delete defunct code

Comment (ConnectedSystems) [reply src/ecosystem/corals/Corals.jl:266]: Oh, sorry - just read your note. Agree we can ignore for a later clean up.

File: src/ecosystem/corals/Corals.jl [hunk 3]
```diff
@@ -369,6 +437,37 @@ function _update_coral_spec(spec::DataFrame, pnames::Vector{String}, coral_param
...
+    _update_coral_factors(spec::DataFrame, coral_params::DataFrame)::DataFrame
+
+Update scenario specification with updated coral parameters/factors.
+Allows changes to `coral_spec()` to be represented without requiring a session restart.
+For debugging purposes only.
+
+# Examples
+```julia
+using Revise
+
```
Comment (ConnectedSystems) [inline src/ecosystem/corals/Corals.jl:450]: Just noting this `using` shouldn't be here

File: src/interventions/seeding.jl [hunk 4]
```diff
@@ -115,3 +115,65 @@ function seed_corals!(
 
     return nothing
 end
+function seed_corals!(
```
[the remark this answers is not in the document]
Comment (ConnectedSystems) [reply src/interventions/seeding.jl:118]: Seems like they handle slightly different argument types for some reason, but they look functionally identical.

Agree with removing the unused one.

File: src/io/datacubes.jl [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+"""
+    copy(cube::YAXArray)::YAXArray
+
+Copy a YAXArray data cube.
+"""
+function Base.copy(cube::YAXArray)::YAXArray
+    new_axlist = Tuple(ax for ax in deepcopy(cube.axes))
+    return YAXArray(new_axlist, copy(cube.data))
+end
```
Comment (ConnectedSystems) [inline src/io/datacubes.jl:108]: Note: technically this is type piracy, we should in fact ask YAXArrays.jl to provide this for us instead.

**Review by ConnectedSystems — APPROVED** [review]
Fine with me

**Pull request merged**